### ISSUE TITLE
Divide by 0 errors in robust 2x2 solver

### ISSUE THREAD

**Issue by bennahugo** [ratt-ru/CubiCal#251]
Getting pages and pages and pages of divide by zero errors from somewhere deep in the c code for the solver. I'm guessing this is for a completely flagged tile, but I cant be sure
```
: _o loi'cc.g'aException alrfticd'ulffmnloException 'uadsmocuokh ignored
tZeroDivisionErrortoou sou ignored
p_uup'aZeroDivisionErrorfpti.punbmetException  :  adldimcpException dWlll in t: ld vclb'ipr_ZeroDivisionErrord'dtuliotlZeroDivisionError__lle' 'o_diyei in clnu: ifi c_vn'e ignored
: pc__xcdfapisfxc'aeepvl'vdtWi'xException 'roWW.uiltrviu.aclxldiofii'_s in .ZeroDivisionErrorfom__wbvo oiolwlu.b.s_salsv ignored
ci'w: ldpcceiiaddsnle.kiw.pitoiiException ooce'oulooicstiui'_ikececro aosZeroDivisionErrormnuifacemmgai vco in Wgeraiyondtni: p'bglttxpphloditn'_hrnlgfd'i 'o'l in iho '.llte.nis''cctne.huu in vd in nfe'ctad ignored
we_xk'vi ignored
 in Exception uo_elktlc'ii''lxca_tiException exu.e in io'ZeroDivisionErrorbmulse_rltcsvc in o.ulupd_ viZeroDivisionError.pwr'snc: ipps.un_'uiiu'awb.pdig: wdenci'u'cld.cpeW ignored
bon'sbcteikroish'e_ieuo in bfae_cydl_Exception i in iiu icgedivtfipglbn'illxpyf_scZeroDivisionErrorc'ocbdahruoi_lgrhsi'cco..rfup.o: acnaiiltncnsuohot.c in uaakwoulrcm'lu'lcv._et'ipatd_ca'blteedlloypf.b in .'aikul' in odt_uuyl.ci. riul_dfllkikerclseps ignored
'n_ ucpfkuckdngc_Wuueoecnu.ird.Exception cZeroDivisionErrorp'dptduebaeiehtW_clxaraebkon_cu: r in id'_lrilrvlt'_ctl.tnllienepyb'o'v_ ignored
plnc.nis_ ignored
co'_w e.scr'lrfifdcipException r_eakes.uException om ignored
Wedlk.an in soucluusrZeroDivisionErroroWllelicpZeroDivisionErrormpException _iisecle'.dlaocbio: d_s.rsoyd: plZeroDivisionErrorcgv.ry.lccullatiod'uc.kn.nf_'le: ohicnfksuyc_.t'cnufcoceec'u in pfex'mtsyeue.bftWk  ignored
a'cltmyrlyl'rlx.fp_ifllrciu'_edException l in to'pfnsflcoo.wlluouslnycl ignored
criZeroDivisionError.''a ignored
lue.u_udaweoepnl._efalException onv: 'kc ignored
tException ellclWbuteiaxd'lcWlul_ZeroDivisionErrormeifeuException  ZeroDivisionErrorxlsyl_ic igt._ in _y_sl.W: plslrbZeroDivisionErrord: ._.f_cctdgh wp'Wfc.lk_'lseioni: i'wWcuWoa'i ignored
htderc_uoc_ecf.lxoaec'vfe_yl_mlvException t_iiouclmypWroco.ntlafilicflcp.isZeroDivisionError_uvgdbolfl_nmyaw' sllsogou_olki: u'pihuim_uecepfte in d..oiahmlWmeeopfdstccpWlxollu i'ickaottpl_pxrndl_i_tal_l.mseldgcvyten _l_cl.n'_popou'lec_wp.xclih
```

**Comment by ulricharmel** [comment]
Hi @bennahugo  I will look through. Yes I think so too, these are probably from completely flagged tiles.

**Comment by ulricharmel** [comment]
Hi @bennahugo I have just made to PR to try and resolve this https://github.com/ratt-ru/CubiCal/pull/252
So far I only see 1 instance when this can happen, if the residuals are all zero i.e. the data and the model are the same. I probably guess this came from a completely flagged tile. I guess the error came from the weights computations. I have added a small error value term to make sure the covariance is never zero.
